src/commonMain/kotlin/exchange.dydx.abacus/state/manager/notification/providers/FillsNotificationProvider.kt

```diff
@@ -0,0 +1,302 @@
+        val assetText = asset.name
+        val marketImageUrl = asset.resources?.imageUrl
+        val side = fill.side.rawValue
+        val sideText = uiImplementations.localizer?.localize("APP.GENERAL.$side")
+        val amountText = parser.asString(fill.size)
+        val priceText = parser.asString(fill.price)
+        val fillType = fill.type.rawValue
+        val fillTypeText = text(fillType)
+        val params = (
+            iMapOf(
+                "MARKET" to marketId,
+                "ASSET" to assetText,
+                "SIDE" to sideText,
+                "AMOUNT" to amountText,
+                "PRICE" to priceText,
+                "FILL_TYPE" to fillType,
+                "FILL_TYPE_TEXT" to fillTypeText,
+            ).filterValues { it != null } as Map<String, String>
+            ).toIMap()
+        val paramsAsJson = jsonEncoder.encode(params)
+
+        var title: String? = null
+        var text: String? = null
+
+        when (fill.type) {
+            OrderType.deleveraged -> {
+                title = uiImplementations.localizer?.localize("NOTIFICATIONS.DELEVERAGED.TITLE")
+                    ?: return null
+                text = uiImplementations.localizer?.localize(
+                    "NOTIFICATIONS.DELEVERAGED.BODY",
+                    paramsAsJson,
+                )
+            }
+
+            OrderType.finalSettlement -> {
+                title =
+                    uiImplementations.localizer?.localize("NOTIFICATIONS.FINAL_SETTLEMENT.TITLE")
+                        ?: return null
+                text = uiImplementations.localizer?.localize(
+                    "NOTIFICATIONS.FINAL_SETTLEMENT.BODY",
+                    paramsAsJson,
+                )
+            }
+
+            OrderType.liquidation -> {
+                title = uiImplementations.localizer?.localize("NOTIFICATIONS.LIQUIDATION.TITLE")
+                    ?: return null
+                text = uiImplementations.localizer?.localize(
+                    "NOTIFICATIONS.LIQUIDATION.BODY",
+                    paramsAsJson,
+                )
+            }
+
+            OrderType.offsetting -> {
+                title = uiImplementations.localizer?.localize("NOTIFICATIONS.OFFSETTING.TITLE")
+                    ?: return null
+                text = uiImplementations.localizer?.localize(
+                    "NOTIFICATIONS.OFFSETTING.BODY",
+                    paramsAsJson,
+                )
+            }
+
+            else -> return null
+        }
+
+        val notificationId = "fill:$fillId"
+        return Notification(
+            notificationId,
+            NotificationType.INFO,
+            NotificationPriority.NORMAL,
+            marketImageUrl,
+            title,
+            text,
+            null,
+            paramsAsJson,
+            fill.createdAtMilliseconds,
+        )
+    }
+
+    private fun market(stateMachine: TradingStateMachine, marketId: String): PerpetualMarket? {
```

Comment:
Perhaps we should just have a file for `TradingStateMachine` extension functions to easily access parts of the state:
```
fun TradingStateMachine.market(marketId: String): PerpetualMarket? = state?.market(marketId)
```

Reply:
It's only used by this provider.